datafusion/core/src/datasource/physical_plan/parquet/access_plan.rs

```diff
@@ -239,7 +245,32 @@ impl ParquetAccessPlan {
             .iter()
             .any(|rg| matches!(rg, RowGroupAccess::Selection(_)))
         {
-            return None;
+            return Ok(None);
+        }
+
+        // validate all Selections
+        for (idx, (rg, rg_meta)) in self
```

Comment:
This is new checking added as users can pass in `ParquetAccessPlan` and the semantics are quite subtle